tensorflow/core/platform/nvtx.h

```diff
@@ -0,0 +1,258 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_PLATFORM_NVTX_H_
+#define TENSORFLOW_CORE_PLATFORM_NVTX_H_
+
+#include "third_party/nvtx3/nvToolsExt.h"
+
+#include "tensorflow/core/framework/attr_value.pb.h"
+#include "tensorflow/core/framework/attr_value_util.h"
+#include "tensorflow/core/framework/types.pb.h"
+#include "tensorflow/core/util/env_var.h"
+
+namespace tensorflow {
+namespace nvtx {
+
+namespace detail {
+
+inline unsigned hash_string(const char* c) {
+  enum { M = 33 };
+  unsigned hash = 5381;
+  while (*c) {
+    hash = hash * M + *c++;
+  }
+  return hash;
+}
+
+inline uint32_t get_color(unsigned hash) {
+  const uint32_t colors[] = {0x00aedb, 0xa200ff, 0xf47835, 0xd41243, 0x8ec127,
+                             0xffb3ba, 0xffdfba, 0xffffba, 0xbaffc9, 0xbae1ff,
+                             0xbbcbdb, 0x9ebd9e, 0xdd855c, 0xf1e8ca, 0x745151,
+                             0x2e4045, 0x83adb5, 0xc7bbc9, 0x5e3c58, 0xbfb5b2,
+                             0xff77aa, 0xaaff77, 0x77aaff, 0xffffff, 0x000000};
+  const int ncolor = sizeof(colors) / sizeof(uint32_t);
```

Comment:
nit: would it be possible to squeeze in 7 more colors (from 25 to 32) so that modulo is by a power of 2?

Reply:
@nluehr which color would u like to support ?